packages/progress-bar/test/progress-bar.test.ts

```diff
@@ -51,6 +47,41 @@ describe('ProgressBar', () => {
 
         expect(el.getAttribute('label')).to.equal('Label From Slot');
     });
+
+    it('handles label attribute changes', async () => {
+        const el = await fixture<ProgressBar>(html`
+            <sp-progress-bar label="label" indeterminate>
+                content
+            </sp-progress-bar>
+        `);
+
+        await elementUpdated(el);
+        el.setAttribute('label', '');
+        await elementUpdated(el);
+
+        expect(el.getAttribute('label')).to.equal('');
+        el.setAttribute('label', 'label1');
+        await elementUpdated(el);
+        expect(el.getAttribute('label')).to.equal('label1');
+    });
+
+    it('renders label when content is absent', async () => {
+        const el = await fixture<ProgressBar>(html`
+            <sp-progress-bar label="myLabel" indeterminate></sp-progress-bar>
+        `);
+        expect(el.getAttribute('label')).to.equal('myLabel');
+    });
+
+    it('renders nothing when both content and label is absent', async () => {
+        const el = await fixture<ProgressBar>(html`
+            <sp-progress-bar label="myLabel" indeterminate></sp-progress-bar>
+        `);
+        el.removeAttribute('label');
+        el.shadowRoot.textContent = '';
+        expect(el.getAttribute('label')).to.equal(null);
+        expect(el.shadowRoot.textContent?.trim()).to.equal('');
```

Comment:
Since, creating an `sp-progress-bar` without label was resulting in accessibility issues , thus initially I have created an `sp-progress-bar` with a label and them I have forcefully removed the label for testing.